moto/awslambda/models.py

```diff
@@ -1205,22 +1167,34 @@ def __init__(self, region_name: str, account_id: str):
         self.region_name = region_name
         self.account_id = account_id
 
+        # function-arn -> alias -> LambdaAlias
+        self._aliases: Dict[str, Dict[str, LambdaAlias]] = defaultdict(lambda: {})
```

Comment:
Aliases are associated with the "lambda" not the "lambda version". We can't store these in "lambda version" as it makes it very complex to check if the alias already exists before writing. It also simplifies fetching a list of aliases as we don't need to iterate over each "version" to find it's alias